script/model.py

```diff
@@ -4,141 +4,350 @@
 import numpy as np
 import torch
 import torch.nn as nn
-import torch.nn.functional as F
-import torch.optim as optim
-from torch.utils.data import Dataset, DataLoader
-
-PI = torch.tensor(np.pi)
-
-class MNIST_Dataset(Dataset):
-    def __init__(self, image):
-        super(MNIST_Dataset).__init__()
-        self.image = image
-    def __len__(self):
-        return self.image.shape[0]
-    def __getitem__(self, idx):
-        image = np.copy(self.image[idx, :].reshape(28,28))
-        image = image.astype(np.float32)
-        direction = np.random.choice(['left', 'right'])
-        image_list = []
-        image_list.append(image.reshape(-1))
-        for k in range(1,20):
-            if direction == 'left':
-                tmp = np.delete(image, 0, 1)
-                image = np.concatenate((tmp, image[:,0].reshape(-1,1)) ,1)
-                image_list.append(image.reshape(-1))
-            elif direction == 'right':
-                tmp = np.delete(image, -1, 1)
-                image = np.concatenate((image[:,-1].reshape(-1,1), tmp) ,1)
-                image_list.append(image.reshape(-1))
-                
-        image_seq = np.array(image_list)
-        return image_seq
+
+
+# nn.Module is a class
+class DBlock(nn.Module):
+    """ A basic building block for parametralize a normal distribution.
```

Comment:
```suggestion
    """ A basic building block for parameterizing a normal distribution.
```

Reply:
Also, I'm not sure if this is true? Is it parameterizing a normal distribution?

Reply:
it's the "re-parameterization  trick." Not sure if the terminology is being used correctly but here is what I found about it. 
https://stats.stackexchange.com/questions/199605/how-does-the-reparameterization-trick-for-vaes-work-and-why-is-it-important